### ISSUE TITLE
Reaching full brightness on H6110, and modifiable brightness curve

### ISSUE THREAD

**Issue by gawtDamn** [bwp91/homebridge-govee#50]
* **Please explain your feature request in a one or two sentences.**
* The Govee H6110 light strip outputs at half power for colors that are close to white when commanded through homebridge vs through the govee App. This is understandable since the RGB value that's commanded is [128,128,128] instead of [255,255,255]. The request is to re-normalize the command before it is issued.

* **Is your feature request related to a problem? Please describe.**
I'd like to submit the following idea- that whatever RGB value is currently being commanded be modified by normalizing as follows: RGB_new = RGB_old * 255 / max(RGB_old) 

This will allow full brightness in the part of the RGB space near white.

* **Any particular Govee devices that this relates to?**
H6110 Light strip

* **Anything else?**

**Comment by bwp91** [comment]
Hi @gawtDamn 

Thank you for your suggestion.

I have incorporated this into a beta version I have just released, v2.11.3-4. I am unable to access my Govee device to see how the colour appears, so your feedback would be great!

**Comment by bwp91** [comment]
I've released another beta, v2.11.3-6, with a change to the formula (one that I tested before).

The issue I was facing was that the colours seemed very white towards the red end of the spectrum, and whilst the blues and greens seemed nice, it was impossible to get a colour close to the 'red' that was possible through the Govee app.

I've gone back to this formula again, with a tweak that lowers the green and blue values if red is 255. And if the greens and blues are so low (<25) then the plugin will reduce these to 0 so the solid red is reachable via the Home app.

Feedback from anyone would be good on this!

**Comment by gawtDamn** [comment]
This works very well (I tested v2.11.3-7) for the issue I reported. You've
got it so that for whatever color, at least one of the RGB values will be
255, and that was the point. The power draw also backs this up, with the
previous peak power draw for my device being 20 watts, and now it is 35
watts.

Thanks!

**Comment by bwp91** [comment]
How are the reds for you? Perhaps it’s just the strip i own or the model i own, but towards the red, the colours seem lighter than they should be.

**Comment by gawtDamn** [comment]
I saw a few times the behavior you're talking about, and I was watching the
homebridge log while it was happening. I used Siri to command red, and the
RGB that showed up in the log was something like [255,124,124] which would
be a light pink. And then when I repeated the command, I got the expected
result, of [255,0,0]. I'm not sure where the pink came from, but repeating
the command was sufficient to fix it.

I did see a related behavior previously using the release version, I think
it was version 2.10 , and I don't understand it. What would happen is I
would command red using Siri, and that would be interpreted as (and
confirmed by Siri as) a command to turn off the light strip. Other colors
would work properly. The way I got this to stop happening was to use one of
the homekit apps (Home, Eve, whichever) and manually select colors near
red. Then after doing this, I could command red using Siri, and it would
actually happen without turning the light strip off.

But at the same time, occasionally, a command for red would result in
[128,128,128], and a command for white would result in [255,0,0]. And this
I don't understand. And I did see this still a time or two on the 2.11.3-7
beta today, just substituting in something more like [255,220,220] for
[128,128,128].

But I haven't looked at the code at all, so I don't know what part of the
commanding process results in the RGB output that makes it into the
homebridge log output, or what is sent to Govee for them to send back to
the light strip, if that's even how that works. And I don't know the form
of the command that Siri passes to homebridge. Quite often I would command
the strip using Siri, and get a confirmation from Siri, but the command
would seem to get lost before it makes it to the device or to the
homebridge log. But every command that showed up in the homebridge log was
actuated on the device.
